src/firefly/java/edu/caltech/ipac/visualize/plot/FitsRead.java

```diff
@@ -1140,11 +1147,17 @@ private static double getPowerLawGammaStretchedPixelValue(double x, double gamma
      *
      *   Since  mu_0 and a are constant, we can use just:
      *     mu =  asinh( flux/(2.0*b)) ) = asin(x/beta); where beta=2*b;
+     * @param flux
+     * @param maxFlux
+     * @param minFlux
      * @param beta
      * @return
      */
     private static double  getASinhStretchedPixelValue(double flux, double maxFlux, double minFlux, double beta)  {
 
+        if (flux <= minFlux) { return 0d; }
+        if (flux >= maxFlux) { return 254d; }
+
```

Comment:
why do our other stretch type not purposely have to do this code as well?

Reply:
other algorithms compute discrete values, all of which are in the range, and return the index of a discrete value, which is closest to the data value